### PR TITLE
Convert Broadcaster to use generics, remove Listenable.

### LINKED ISSUE #1604: Broadcaster is public, and only one per object
Currently, `Broadcaster` requires that a `Listenable` be passed in, and a `Listenable` has the following contract:

```Typescript
export interface Listenable {
  broadcaster: Broadcaster;
}
```

This setup has a number of problems:
* The `broadcaster` field is public.
* Having more than one `Broadcaster` per `Listenable` is difficult, because one has to be named "`broadcaster`".

Instead of defining `Listenable` as a distinct type, we should allow `Broadcaster`s to be able to be attached to arbitrary `Object`s.

### PR DESCRIPTION
**Pull request by jtlan**
Previously: `Broadcasters` had-a `Listenable`, which had-a reference to the same `Broadcaster`. This was silly and prevented the use of multiple
`Broadcasters` on the same object.

Now we can use generic typing to convey more information about what the `Broadcaster` is broadcasting.

Close #1604.

### CODE REVIEW

File: src/core/broadcaster.ts [hunk 1]
```diff
@@ -3,50 +3,27 @@
...
-    public listenable: Listenable;
+    private _listenable: L;
 
     /**
-     * Constructs a broadcaster, taking the Listenable that the broadcaster will be attached to.
+     * Constructs a broadcaster, taking a "listenable" object to broadcast about.
      *
      * @constructor
-     * @param {Listenable} listenable The Listenable-object that this broadcaster is attached to.
+     * @param {L} listenable The listenable object to broadcast about.
```
Comment (bluong) [inline src/core/broadcaster.ts]: broadcast about sounds a bit odd...  Is that right semantically?

File: src/core/broadcaster.ts [hunk 2]
```diff
@@ -55,10 +32,10 @@ export module Core {
      * If there is already a callback associated with that key, then the callback will be replaced.
      *
      * @param key The key associated with the callback. Key uniqueness is determined by deep equality.
-     * @param {BroadcasterCallback} callback A callback to be called when the Scale's domain changes.
-     * @returns {Broadcaster} this object
+     * @param {(listenable: L, ...args: any[]) => any} callback A callback to be called.
```
Comment (bluong) [inline src/core/broadcaster.ts]: I assume type doesn't work?  Maybe just use an interface here to better inform / encapsulate

File: src/core/resizeBroadcaster.ts [hunk 3]
```diff
@@ -14,12 +14,12 @@ export module Core {
...
    */
   export module ResizeBroadcaster {
-    var broadcaster: Core.Broadcaster;
+    var broadcaster: Core.Broadcaster<any>;
     var _resizing: boolean = false;
 
     function _lazyInitialize() {
       if (broadcaster === undefined) {
-        broadcaster = new Core.Broadcaster(<any> ResizeBroadcaster);
+        broadcaster = new Core.Broadcaster<any>(<any> ResizeBroadcaster);
```
Comment (bluong) [inline src/core/resizeBroadcaster.ts]: What is with the casting here

**Comment by bluong** [comment]
Very minor issues.

File: src/core/broadcaster.ts (hunk 1 above)
Comment (jtlan) [inline src/core/broadcaster.ts]: maybe just "to broadcast"?

File: src/core/broadcaster.ts (hunk 2 above)
Comment (jtlan) [inline src/core/broadcaster.ts]: Can't define the interface inside the class either, and the generic isn't scoped outside of it.

File: src/core/broadcaster.ts (hunk 1 above)
Comment (bluong) [inline src/core/broadcaster.ts]: Works for me

File: src/core/resizeBroadcaster.ts (hunk 3 above)
Comment (jtlan) [inline src/core/resizeBroadcaster.ts]: The first `<any>` is the generic. The second `<any>` was a cast that isn't needed anymore.

File: src/core/broadcaster.ts (hunk 2 above)
Comment (bluong) [inline src/core/broadcaster.ts]: We however can define a generic typed interface at the module level and pass in the L.  Would that work?

**Pull request merged**